plugin/pkg/scheduler/algorithm/predicates/predicates.go

```diff
@@ -196,15 +205,25 @@ type VolumeFilter struct {
 // the maximum.
 func NewMaxPDVolumeCountPredicate(filter VolumeFilter, maxVolumes int, pvInfo PersistentVolumeInfo, pvcInfo PersistentVolumeClaimInfo) algorithm.FitPredicate {
 	c := &MaxPDVolumeCountChecker{
-		filter:     filter,
-		maxVolumes: maxVolumes,
-		pvInfo:     pvInfo,
-		pvcInfo:    pvcInfo,
+		filter:                filter,
+		maxVolumes:            maxVolumes,
+		pvInfo:                pvInfo,
+		pvcInfo:               pvcInfo,
+		randomVolumeIDPrefix:  rand.String(8),
+		volumeIDPrefixCounter: 1,
 	}
 
 	return c.predicate
 }
 
+func (c *MaxPDVolumeCountChecker) getNextVolumeIDPrefix() string {
+	c.lock.Lock()
+	defer c.lock.Unlock()
```

Comment:
can be lock-free with `atomic.AddInt64(&c.volumeIDPrefixCounter, 1)`

Reply:
Smooth! Changed it.